indico/modules/rb_new/client/js/modules/calendar/reducers.js

```diff
@@ -271,5 +272,17 @@ export default combineReducers({
                 return state;
         }
     },
-    datePicker: datePickerReducer
+    datePicker: datePickerReducer,
+    view: (state = 'timeline', action) => {
+        switch (action.type) {
+            case calendarActions.CHANGE_VIEW: {
+                if (['timeline', 'list'].includes(action.view)) {
```

Comment:
I don't think reducers need to validate the incoming data. That can be done in the query string rule using this:

    validator: v.isIn(['timeline', 'list']),